client/cli/plop/utils.ts

```diff
@@ -0,0 +1,17 @@
+import fetch from 'node-fetch';
+
+import { PackageMetadata } from './typings';
+
+const getDependencyMetadata = (name: string) =>
+  fetch(`https://registry.npmjs.org/${name}`, {
+    headers: { Accept: 'application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*' }
+  }).then(r => r.json());
+
+export const getVersionsForDependencies = async (deps: string[]) =>
+  (await Promise.all(deps.map(d => getDependencyMetadata(d)))).reduce(
```

Comment:
Also here do we have to use async await?
Returning promise is enough imo

Reply:
So I can ‘reduce’ on the promise.all result to a name:version object